### PR TITLE
Merge cobaya/master

### PR DESCRIPTION
**Pull request by AdamOrmondroyd**
The cobaya master branch has gotten well ahead of this fork, this PR attempts to bring the two back together.

This came to my attention as the MCMC sampler is now broken on this fork as of pandas 2, and while this fork is intended to make PolyChord work properly, it would be daft to require reinstalling just to use MCMC. 

I suggest having a very close look through cobaya/samplers/polychord/polychord.py, which was the non-trivial merge

### CODE REVIEW

**Comment by AdamOrmondroyd** [comment]
@williamjameshandley please could you have a look at this asap? I'll try keep on top of further updates after this one.

I've noticed the DES likelihoods take an awful lot longer to run after these changes.

File: cobaya/samplers/polychord/polychord.py [hunk 1]
```diff
@@ -142,11 +154,9 @@ def initialize(self):
                    "cluster_posteriors", "write_resume", "read_resume",
                    "write_stats", "write_live", "write_dead", "write_prior",
                    "maximise", "compression_factor", "synchronous", "base_dir",
-                   "file_root", "seed", "grade_dims", "grade_frac", "nlives"]
+                   "file_root", "grade_dims", "grade_frac", "nlives"]
```
Comment (williamjameshandley) [inline cobaya/samplers/polychord/polychord.py:157]: Is there a reason seed has been removed?

**Review by williamjameshandley — COMMENTED** [review]
Other than the removal of seed this looks like it's cobaya-level changes rather than anything polychord api related

File: cobaya/samplers/polychord/polychord.py (hunk 1 above)
Comment (williamjameshandley) [reply cobaya/samplers/polychord/polychord.py:157]: Does this mean this closes #3?

**Review by williamjameshandley — APPROVED** [review]

File: cobaya/samplers/polychord/polychord.py (hunk 1 above)
Comment (AdamOrmondroyd) [reply cobaya/samplers/polychord/polychord.py:157]: yes

**Pull request merged**